packages/aws-rfdk/lib/core/scripts/bash/installLustreClient.sh

```diff
@@ -0,0 +1,288 @@
+#!/bin/bash
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script downloads and installs the Lustre client.
+# See https://docs.aws.amazon.com/fsx/latest/LustreGuide/install-lustre-client.html
+
+set -xeu
+shopt -s extglob
+
+trap exit_trap EXIT
+function exit_trap() {
+  if [[ $? -ne 0 ]]; then
+    echo "ERROR: An error occurred while attempting to install the Lustre client on your device. \
+    Please refer to the FSx for Lustre documentation: https://docs.aws.amazon.com/fsx/latest/LustreGuide/install-lustre-client.html"
+  fi
+}
+
+function is_kernel_version_lower() {
+  local IFS='.-'
+  read -r -a lhs_array <<< "$1"
+  read -r -a rhs_array <<< "$2"
+
+  # Calculate minimum length betweeen the two arrays
+  [[ ${#lhs_array[@]} -lt ${#rhs_array[@]} ]] && n=${#lhs_array[@]} || n=${#rhs_array[@]}
+
+  unset IFS
+  for i in $( seq 0 $((n-1)) ); do
+    local lhs="${lhs_array[$i]}"
+    local rhs="${rhs_array[$i]}"
+    if [[ $lhs =~ ^[0-9]+$ && $rhs =~ ^[0-9]+$ && $lhs -lt $rhs ]]; then
+      return 0
+    fi
+  done
+  return 1
+}
+
+function install_on_al2() {
+  architecture=$(uname -m)
+  case $architecture in
+    x86_64)   min_version="4.14.104-95.84.amzn2.x86_64";;
+    aarch64)  min_version="4.14.181-142.260.amzn2.aarch64";;
+    *) echo "ERROR: Unrecognized Amazon Linux 2 kernel version: $kernel_version"; exit 1;;
+  esac
+
+  kernel_version=$(uname -r)
+  if is_kernel_version_lower "$kernel_version" "$min_version"; then
+    echo "ERROR: Kernel version ($kernel_version) is lower than the minimum required version: $min_version"
+    exit 1
+  fi
+
+  sudo amazon-linux-extras install -y lustre2.10
+}
+
+function install_on_al1() {
+  min_version="4.14.104-78.84.amzn1.x86_64"
+
+  kernel_version=$(uname -r)
+  if is_kernel_version_lower "$kernel_version" "$min_version"; then
+    echo "ERROR: Kernel version ($kernel_version) is lower than the minimum required version: $min_version"
+    exit 1
+  fi
+
+  sudo yum install -y lustre-client
+}
+
+function verify_lustre_kmod() {
+  modprobe -v --first-time lustre
+  if [[ $? -ne 0 ]]; then
```

Comment:
The shell script uses `set -e`, so if the `modprobe` doesn't return 0 then we'll bail immediately and never get to this condition test, no?

i.e. Should these two lines instead be?
```
if modprobe -v --first-time lustre; then
```
```